### PR TITLE
Wrangler and C3: Move the cli folder of C3 into @cloudflare/cli and make Wrangler and C3 depend on it

### PR DESCRIPTION
**Pull request by gabivlj**
**What this PR solves:**
We want to start using @clack/core and C3-like components in Wrangler, so the logical next step is to make these components available to wrangler and C3.

1. ./helpers/cli.ts in C3 is now index.ts in @cloudflare/cli, didn't include `openInBrowser`,`C3_DEFAULTS` and `WRANGLER_DEFAULTS`.
2. ./helpers/colors in C3 is now colors.ts in @cloudflare/cli.
3. ./helpers/interactive.ts in C3 is now interactive.ts in @cloudflare/cli.

- [x] Tests (a basic test)
- [x] Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))

**Reviewer is to perform the following, as applicable:**

- Checked for inclusion of relevant tests
- Checked for inclusion of a relevant changeset
- Checked for creation of associated docs updates
- Manually pulled down the changes and spot-tested

**Note for PR author:**

We want to celebrate and highlight awesome PR review! If you think this PR received a particularly high-caliber review, please assign it the label `highlight pr review` so future reviewers can take inspiration and learn from it.

### CODE REVIEW

File: packages/cli/__tests__/cli.test.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { describe, expect, test } from "vitest";
```
Comment (jculvey) [inline packages/cli/__tests__/cli.test.ts:1]: If we want this to run in CI I think we'll have to add a workflow to `.github/workflows`

File: packages/cli/.eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+	root: true,
+	extends: ["@cloudflare/eslint-config-worker"],
+	ignorePatterns: ["dist", "scripts"],
```
Comment (jculvey) [inline packages/cli/.eslintrc.js:4]: I'm guessing we don't need `scripts` anymore here

**Review by jculvey — APPROVED** [review]
Changes LGTM. I've been intending to revisit some of the naming in `interactive.ts` for a while, which I'd still like to do at some point. I'm curious to hear more about your use case, perhaps we can chat some time.

**Comment by gabivlj** [comment]
> Changes LGTM. I've been intending to revisit some of the naming in `interactive.ts` for a while, which I'd still like to do at some point. I'm curious to hear more about your use case, perhaps we can chat some time.

Sure! We also want to introduce some changes we did where we added a multiselect component, perhaps we can discuss those naming changes in that future PR and talk more about our interactivity use-cases.

File: packages/cli/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+		"target": "ESNext",
+		"module": "CommonJS",
+		"moduleResolution": "node",
+		"allowJs": false,
+		"strict": true,
+		"esModuleInterop": true,
+		"outDir": "dist",
+		"types": ["node"],
+		"resolveJsonModule": true
+	}
```
[the remark this answers is not in the document]
Comment (gabivlj) [reply packages/cli/tsconfig.json]: Updated to keep it barebones

File: .github/workflows/cli.yml
[the remark this answers is not in the document]
Comment (gabivlj) [reply .github/workflows/cli.yml:1]: Sure! Just added it because of the suggestion above. Anyway I think that makes sense.

**Comment by gabivlj** [comment]
Rebased

**Comment by gabivlj** [comment]
Rebased again

**Review by penalosa — APPROVED** [review]

**Pull request merged**